proposals/dra-mode.md

```diff
@@ -0,0 +1,127 @@
+use, it does overlap a lot with it.
+
+This new system is relevant to Akri as it would ease the management of slots usage and shared devices, for both instance level and configuration level resources.
+
+As a quick reference, the DRA uses a set of kubernetes resources to work: A ResourceClass that register a new class of devices that might be allocatable
+(with parameters for the driver) and a ResourceClaim that describe a request for allocation of a resource (with parameters for the driver).
+These resources all ultimately refer to a driver, that is a two components system comprised of a resource controller and a resource kubelet plugin.
+The controller will help the scheduler on node choice, and receive (de)allocation requests for the devices. The kubelet plugin will give the kubelet
+all needed information to expose the device to the workload for a given ResourceClaim. The tracking of available resources is completely up to the driver.
+
+In comparison, the current Akri model revolves around Configurations and Instances. A configuration both describe how to discover devices and what to
+schedule when they are discovered (broker logic). An instance just represent a discovered instance and the list of nodes where it is available.
+
+This proposal primarily focus on the Configuration resource, as the Instance resource is all about available resource tracking that is explicitly not
+covered by DRA.
+
+The Configuration resource match with the ResourceClass for the discovery part, the ResourceClaim part is handled by a combination of the Instance and
+inner workings of the akri agent.
+
+## API Changes
+
+These API changes may require writing a migration tool for existing Configurations
+
+### Create a `BrokerTemplate` resource kind
+
+In order to fit in the DRA model we need to separate the broker logic from the discovery logic, thus we need to create a new resource kind to handle the broker part. This resource needs to be linked to a discovery logic resource and provides all the needed information about what to schedule.
+
+This uses the "Deploy arbitrary Kubernetes resources as broker" for the description of what to schedule, please see that proposal for further details
+on the exact content of the broker fields.
+
+```yaml
+# apiVersion got ommited on purpose
+kind: BrokerTemplate
+metadata: ...
+spec:
+    discoveryReference: foo
+    instanceBrokerResources: ...
+    globalBrokerResources: ... # this field is named configurationBrokerResources in other proposal
+```
+
+### Create a `DiscoveryConfiguration` resource kind
+
+This resource bears the discovery part of the current Configuration resource. It would be used as ResourceClass parameter.
+You can note that we do **not** include the name of the discovery handler as it would be in the driver name in ResourceClass
+as a subdomain of `driver.akri.sh` like e.g. `udev.driver.akri.sh`
```

Comment:
I wonder if that's the best idea to have a driver name per DH, maybe having a single `driver.akri.sh` DRA driver name is better and thus re-introduce discovery handler name in the `DiscoveryConfiguration`, as it would remove any ambiguity about what DH the configuration is for.

Another idea here was to have a dedicated CRD per DH so that way there is no ambiguity on what DH a configuration is for, but it would make creating a DH more cumbersome as one would also need to create a CRD. It would also make it harder for the agent as it would need to extract information from a possibly non akri CRD, hence I did not pursued on this idea.

Reply:
Another argument against different CRDs is that the ability to tie things together with downwards api seems useful here and an increase in resource types could make building higher level tooling harder (UI for instance)